homeassistant/components/template/weather.py

```diff
@@ -452,3 +468,252 @@ def _validate_forecast(
                 )
             continue
         return result
+
+
+@dataclass
```

Comment:
This has quite a lot of attributes, I'd suggest to make the constructor keyword-only:
```suggestion
@dataclass(kw_only=True)
```